### PR TITLE
Change time comparisons to exclusive by default, add `--time-inclusive`

### LINKED ISSUE #794: --change-newer-than requires using UTC time, "newer" check is incorrect
# Describe the bug you encountered

It looks like there's differences to `--change-newer-than`' s behavior vs. its documentation, as well as having a potential bug.

## Bug 1: Documentation of UTC time usage

It appears that it needs the time to be stated in UTC, not local time.

```
$ touch a

$ sleep 61

$ touch b

$ ls -l a b
-rw-r--r--  1 zachriggle  staff  12168 Jul 15 19:36 a
-rw-r--r--  1 zachriggle  staff      0 Jul 15 19:37 b

$ date -r b '+%Y-%m-%d %H:%M:%S'
2021-07-15 19:37:01

$ fd --change-newer-than '2021-07-15 19:37:01'
a
b
```

However, if we add the `-u` flag to BSD `date` to get UTC time, we get closer to the expected result.

## Bug 2: Files not actually "newer" are printed

Even if we specify in UTC time, file `b` is still printed.

```
$ date -ur b '+%Y-%m-%d %H:%M:%S'
2021-07-16 00:37:01

$ fd --change-newer-than '2021-07-16 00:37:01'
b
```

This is probably technically correct, since the file was created sometime within that second -- but I consider this to be a bug.

However, if I specify the time in seconds, the comparison should be of the same granularity.

## Bug 3: Time comparison seems to be incorrect

Let's try with fractions of a second.  Luckily, GNU `date` does this (macOS default `date` does not) and is available via homebrew.

```
$ gdate -r b +"%Y-%m%-d %H:%M:%S.%N"
2021-0715 19:37:01.866605000

$ fd --change-newer-than '2021-07-16 00:37:01.866605000'
b
```

So it looks like `fd` supports fractions of a second, but is still flagging file `b` as being "newer".  Let's adjust the time by a tiny fraction of a second.

```
$ fd --change-newer-than '2021-07-16 00:37:01.866605001'
<no output>
```

This is the desired output, but it took a long time to get here.

# Describe what you expected to happen:

No files should show up, since the date was extracted from the newer file's timestamp.

# What version of `fd` are you using?

```
$ fd --version
fd 8.2.1
```

**Which operating system / distribution are you on?**

```
$ uname -srm
Darwin 21.0.0 x86_64
```

### PR DESCRIPTION
**Pull request by jacobmischka**
As the names imply, `--change-newer-than` and `--change-before` now are
exclusive (`>` or `<`) by default. A new flag `--time-inclusive`
(aliased to simply `--inclusive`) is added to opt into the previous
inclusive (`>=` or `<=`) behavior.

Fixes #794

### CODE REVIEW

File: doc/fd.1 [hunk 1]
```diff
@@ -217,7 +217,9 @@ tebibytes
 .TP
 .BI "\-\-changed-within " date|duration
 Filter results based on the file modification time.
-Files with modification times greater than or equal to the argument will be returned.
+Files with modification times greater the argument will be returned.
```
Comment (tmccombs) [inline doc/fd.1:246]: I don't think the `than` should be removed from this sentence.

```suggestion
Files with modification times greater than the argument will be returned.
```

**Review by tmccombs — CHANGES_REQUESTED** [review]
I'm also not entirely sure if we need the `--time-inclusive` option.

**Comment by sharkdp** [comment]
I haven't followed the discussion in #794 closely. Can the bug be fixed without introducing a new command-line option? @jacobmischka are you still interested to work on this?

**Comment by tmccombs** [comment]
> Can the bug be fixed without introducing a new command-line option? 

I think so. The new command line option is if we want to keep a way to use an inclusive range, but I'm not really sure if that is needed.

**Comment by jacobmischka** [comment]
Sure, though if we don't want to keep the existing behavior with a flag this PR is probably not worth revising, since all that's left then is to remove two `=` characters. I can do that, if that's the route we want to go.

**Comment by sharkdp** [comment]
> Sure, though if we don't want to keep the existing behavior with a flag this PR is probably not worth revising, since all that's left then is to remove two `=` characters. I can do that, if that's the route we want to go.

I would prefer this. Any help would be very much appreciated.

**Comment by jacobmischka** [comment]
I made a new PR since it was frankly just easier to start from master. #886

**Pull request closed without merging**